pkg/identity/gitlabcom/principal_test.go

```diff
@@ -109,7 +135,46 @@ func TestJobPrincipalFromIDToken(t *testing.T) {
 				"ref_type":        "branch",
 			},
 			WantErr:     true,
-			ErrContains: "project_path",
+			ErrContains: "ci_config_ref_uri",
+		},
+		`Token missing ci_config_sha claim is ok`: {
```

Comment:
Could you clarify why we allow this to be empty?

Reply:
I believe this might not always be populated if the ci_config_ref_uri is pointing to a config outside the current project/repository.

Reply:
> I believe this might not always be populated if the ci_config_ref_uri is pointing to a config outside the current project/repository.

That's correct. Also, adding the new `ci_config_sha` claim to our ID tokens is still being rolled out separately so it may not be available on gitlab.com just yet: https://gitlab.com/gitlab-org/gitlab/-/merge_requests/124647